src/hocon_scanner.xrl

```diff
@@ -50,11 +50,14 @@ Float               = {Integer}?{Fraction}|{Integer}{Fraction}{Exponent}
 
 %% String
 Hex                 = [0-9A-Fa-f]
-Escape              = ["\\bfnrt]
+EscapeNoQuote       = [\\bfnrt]
+Escape              = "|{EscapeNoQuote}
 UnicodeEscape       = u{Hex}{Hex}{Hex}{Hex}
 Char                = ([^\"{LineFeed}]|\\{Escape}|\\{UnicodeEscape})
 String              = "{Char}*"
-MultilineChar       = ([^"]|"[^"]|""[^"]|\\{Escape}|\\{UnicodeEscape})
+%% Special handling for trailing quote: if we don't assert it's not followed by two other
+%% quotes, `{Escape}` would "eat" one of the quotes in the triple quote...
```

Comment:
could you help to explain why this "eat" happens?

This should work?
```
([^"]|"[^"]|""[^"]|\\{EscapeNoQuote}|\\{UnicodeEscape})
```

Reply:
The eat happens because the dangling backslash is the last thing the triple quoted string.  `...\\"""` essentially eats the last quote and breaks out of the triple quote.

```
"""aaa\"""\nx = 10\ny = """~...
```

Since `\` was not being escaped when "unparsing" the config, it would eat the triple quote, and everything after that gets interpreted as the string until it hits another triple quote.

It's easy to see if you try the test case out without the code changes.

Reply:
I thought I had tried your suggestion before, but paired with other changes (escaping or not escaping the string, various combinations of changes), but didn't work before, but now the added test case works with that regex.  :thinking: